src/store/filestore.rs

```diff
@@ -49,206 +43,196 @@ impl Store for LocalFileStore {
 
-        fs::remove_file(self.make_annotation_path::<T>(&hash, name, version))?;
         Ok(())
     }
 }
 
 impl LocalFileStore {
-    /// New function that takes the directory as where to save the files
+    /// Construct a local file store instance in a specific directory.
     pub fn new(directory: impl AsRef<Path>) -> Self {
         Self {
             directory: directory.as_ref().into(),
         }
     }
-
-    /// Getter function for directory
+    /// Get the directory where store is located.
     pub fn get_directory(&self) -> &Path {
         &self.directory
     }
-
-    fn make_dir_path<T>(&self, hash: &str) -> PathBuf {
+    /// Relative path where model specification is stored within the model directory.
+    pub const SPEC_RELPATH: &str = "spec.yaml";
+    /// Relative path where model annotation is stored within the model directory.
+    pub fn make_annotation_relpath(name: &str, version: &str) -> PathBuf {
+        PathBuf::from(format!("annotation/{name}-{version}.yaml"))
+    }
+    /// Build the storage path with the model directory (`hash`) and a file's relative path.
+    pub fn make_path<T>(&self, hash: &str, relpath: impl AsRef<Path>) -> PathBuf {
         PathBuf::from(format!(
             "{}/{}/{}",
             self.directory.to_string_lossy(),
             get_type_name::<T>(),
-            hash,
+            hash
         ))
+        .join(relpath)
+    }
+
+    fn parse_annotation_path(path: &Path) -> Result<impl Iterator<Item = Result<ModelInfo>>> {
+        let re = Regex::new(
+            r"(?x)
+            ^.*
+            \/(?<class>[a-z_]+)
+                \/(?<hash>[0-9a-f]+)
+                    \/annotation
+                        \/
+                        (?<name>[0-9a-zA-Z\-]+)
+                        -
+                        (?<version>[0-9]+\.[0-9]+\.[0-9]+)
+                        \.yaml
+            $",
+        )?;
+        let paths = glob::glob(&path.to_string_lossy())?.map(move |filepath| {
+            let filepath_string = String::from(filepath?.to_string_lossy());
+            let group = re
+                .captures(&filepath_string)
+                .ok_or_else(|| OrcaError::from(Kind::NoRegexMatch))?;
+            Ok(ModelInfo {
+                name: group["name"].to_string(),
+                version: group["version"].to_string(),
+                hash: group["hash"].to_string(),
+            })
+        });
+        Ok(paths)
     }
 
-    /// Helper function for making path to a given item type T
-    pub fn make_path<T>(&self, hash: &str, file_name: &str) -> PathBuf {
-        self.make_dir_path::<T>(hash).join(file_name)
+    fn lookup_hash<T>(&self, name: &str, version: &str) -> Result<String> {
+        let model_info = Self::parse_annotation_path(
+            &self.make_path::<T>("*", &Self::make_annotation_relpath(name, version)),
+        )?
+        .next()
+        .ok_or_else(|| {
+            OrcaError::from(Kind::NoAnnotationFound(
```

Comment:
This relates to the point of whether we want no annotation model to be possible

Reply:
Actually, this is a bit different. This case happens when a user has tried to either `delete_model`, `delete_annotation`, or `load_model` specifically with the `Annotation` option but the `(name, version)` they supplied does not exist in the store i.e. invalid user input.